apps/emqx_modules/src/emqx_telemetry.erl

```diff
@@ -408,7 +408,7 @@ vm_specs() ->
     SysMemData = memsup:get_system_memory_data(),
     [
         {num_cpus, erlang:system_info(logical_processors)},
-        {total_memory, proplists:get_value(available_memory, SysMemData)}
+        {total_memory, proplists:get_value(total_memory, SysMemData)}
```

Comment:
I was thinking if we should try to use `available_memory` first, if available, and fallback to `total_memory` if there is no such key.  From what I understood from the docs, the `available_memory` is more accurate when that info is available in newer kernels.

Maybe something like:

```erlang
  TotalMemory = case proplists:get_value(available_memory, SysMemData) of
                    undefined ->
                        proplists:get_value(total_memory, SysMemData);
                    Mem ->
                        Mem
                end
```

What do you think?

Reply:
🤔 What we intend to show user as total_memory? I mean, how are they supposed to use it?
As far as I understand, available_memory does not include busy memory (i.e. really occupied by processes), is this the intended behavior?

Reply:
I also thought that available_memory changes over time, while cpu number is not, so I  supposed that we would like to provide some "constant initial" machine resources

Reply:
telemetry is not for users.
it's for us to collect the deployment/runtime info to gain insights on what kind of environments EMQX is being deployed in and running in.

runtime information is less important than deployment information. i.e. I'd suggest we report the more "static" data, which is total memory in this case.

Reply:
> I also thought that available_memory changes over time, while cpu number is not, so I supposed that we would like to provide some "constant initial" machine resources

Indeed, I have confused myself.  I guess `total_memory` or maybe `system_total_memory` does make more sense.  Ignore me.  :smile_cat: